vsphere/datadog_checks/vsphere/vsphere.py

```diff
@@ -99,6 +101,8 @@ def initiate_api_connection(self):
             self.api = VSphereAPI(self._config, self.log)
             self.log.debug("Connected")
         except APIConnectionError:
+            # Clear the API connection object if the authentication fails
+            self.api = cast(VSphereAPI, None)
```

Comment:
I'm wondering what kind of effect this will have. Ideally we should handle re-creating the api with smart-retry https://github.com/DataDog/integrations-core/blob/master/vsphere/datadog_checks/vsphere/api.py#L24 https://github.com/DataDog/integrations-core/pull/10101/files